src/main/java/com/ably/kafka/connect/client/DefaultAblyBatchClient.java

```diff
@@ -118,17 +226,48 @@ protected boolean shouldSkip(SinkRecord record) {
      * Function that uses the Ably REST API
      * to send records in batches.
      * @param batches
+     * @return true if non-retriable, false otherwise
      * @throws AblyException
      */
-    public void sendBatches(final List<BatchSpec> batches) throws AblyException {
+    public HttpPaginatedResponse sendBatches(final List<BatchSpec> batches) throws AblyException {
         final HttpCore.RequestBody body = new HttpUtils.JsonRequestBody(batches);
-        final Param[] params = new Param[] { new Param("newBatchResponse", "true") };
+        final Param[] params = new Param[]{new Param("newBatchResponse", "true")};
         final HttpPaginatedResponse response =
                 this.restClient.request("POST", "/messages", params, body, null);
         logger.info(
                 "Response: " + response.statusCode
                         + " error: " + response.errorCode + " - " + response.errorMessage
         );
 
+        return response;
+    }
+
+    /**
+     * Function to parse the able response message
+     * and retrieve the list of failed channel ids.
+     * @param element
+     */
+    public Set<String> getFailedChannels(JsonElement element) {
+
+        Set<String> failedChannels = new HashSet<>();
+
+        // JsonElement element = JsonParser.parseString(errorMessage);
+        // int successCount = element.getAsJsonObject().getAsJsonPrimitive("successCount").getAsInt();
```

Comment:
Is this commented-out code no-longer needed? Can it be removed?

Reply:
Removed.